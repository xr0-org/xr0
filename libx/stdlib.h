#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
