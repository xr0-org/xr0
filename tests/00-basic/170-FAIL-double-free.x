#include <stdlib.h>

void
unit()
{
	void *p; void *q;

	p = malloc(1);
	q = p;
	free(q);
	free(p);
}
