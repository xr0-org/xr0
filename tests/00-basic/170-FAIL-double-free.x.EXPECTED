stdlib.h:10:31: double free (free)
	00-basic/170-FAIL-double-free.x:11:9 (unit)
